### PR TITLE
Send cell centered fluxes with FD.

### PR DESCRIPTION
**Pull request by nilsdeppe**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/DgSubcell/CellCenteredFlux.hpp [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                       subcell::Tags::SubcellOptions<Dim>,
+                       subcell::Tags::Mesh<Dim>, subcell::Tags::DidRollback>;
+
+  template <typename... FluxTags, typename... Args>
+  static void apply(
+      const gsl::not_null<std::optional<Variables<tmpl::list<FluxTags...>>>*>
+          cell_centered_fluxes,
+      const subcell::SubcellOptions& subcell_options,
+      const Mesh<Dim>& subcell_mesh, const bool did_rollback, Args&&... args) {
+    if ((ComputeOnlyOnRollback and did_rollback) or not ComputeOnlyOnRollback) {
```
Comment (kidder) [inline src/Evolution/DgSubcell/CellCenteredFlux.hpp]: this is equivalent to `if(did_rollback or not ComputeOnlyOnRollback)`

File: src/Evolution/DgSubcell/CellCenteredFlux.hpp [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  using return_tags =
+      tmpl::list<subcell::Tags::CellCenteredFlux<flux_variables, Dim>>;
+  using argument_tags =
+      tmpl::push_front<typename FluxMutator::argument_tags,
+                       subcell::Tags::SubcellOptions<Dim>,
+                       subcell::Tags::Mesh<Dim>, subcell::Tags::DidRollback>;
+
+  template <typename... FluxTags, typename... Args>
+  static void apply(
+      const gsl::not_null<std::optional<Variables<tmpl::list<FluxTags...>>>*>
```
Comment (kidder) [inline src/Evolution/DgSubcell/CellCenteredFlux.hpp:46]: include for gsl::not_null

File: src/Evolution/DgSubcell/CellCenteredFlux.hpp [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+#pragma once
+
+#include <cstddef>
+#include <optional>
+#include <utility>
+
+#include "DataStructures/DataBox/PrefixHelpers.hpp"
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
```
Comment (kidder) [inline src/Evolution/DgSubcell/CellCenteredFlux.hpp]: don't think you need this one

File: tests/Unit/Evolution/DgSubcell/Test_CellCenteredFlux.cpp [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+          derivative_order != 0 ? std::optional<size_t>{derivative_order}
+                                : std::optional<size_t>{}},
+      mesh, typename CellCenteredFluxTag::type{},
+      Variables<flux_variables>{mesh.number_of_grid_points(), 1.0});
+
+  db::mutate_apply<evolution::dg::subcell::fd::CellCenteredFlux<
+      TestSystem, Fluxes<Dim>, Dim, ComputeOnlyOnRollback>>(
+      make_not_null(&box));
+  if (derivative_order > 2 and
+      (not ComputeOnlyOnRollback or (ComputeOnlyOnRollback and did_rollback))) {
```
Comment (kidder) [inline tests/Unit/Evolution/DgSubcell/Test_CellCenteredFlux.cpp]: same simplification as above

**Review by kidder — CHANGES_REQUESTED** [review]

**Comment by nilsdeppe** [comment]
Done, rebased, squashed. Thanks for the review :D

**Review by kidder — DISMISSED** [review]

**Comment by nilsdeppe** [comment]
Sorry, fixed another include and squashed...

**Review by kidder — APPROVED** [review]

**Pull request merged**